src/install.c

```diff
@@ -410,33 +410,41 @@ GList* get_install_images(const RaucManifest *manifest, GHashTable *target_group
 	return install_images;
 }
 
-static void parse_handler_output(gchar* line)
+static gchar* parse_handler_output(gchar* line)
 {
+	gchar *message = NULL;
 	g_auto(GStrv) split = NULL;
 
 	g_assert_nonnull(line);
 
 	if (!g_str_has_prefix(line, "<< ")) {
 		g_print("# %s\n", line);
-		return;
+		return NULL;
 	}
 
 	split = g_strsplit(line, " ", 5);
 
 	if (!split[1])
-		return;
+		return NULL;
 
 	if (g_strcmp0(split[1], "handler") == 0) {
-		g_print("Handler status: %s\n", split[2]);
+		message = g_strdup_printf("Handler status: %s", split[2]);
 	} else if (g_strcmp0(split[1], "image") == 0) {
-		g_print("Image '%s' status: %s\n", split[2], split[3]);
-	} else if (g_strcmp0(split[1], "error") == 0) {
-		g_print("error: '%s'\n", split[2]);
+		message = g_strdup_printf("Image '%s' status: %s", split[2], split[3]);
 	} else if (g_strcmp0(split[1], "bootloader") == 0) {
-		g_print("error: '%s'\n", split[2]);
+		message = g_strdup_printf("Bootloader status: %s", split[2]);
+	} else if (g_strcmp0(split[1], "error") == 0) {
+		g_autofree gchar *joined = g_strjoinv(" ", &split[2]);
+		message = g_strdup_printf("Error: %s", joined);
+	} else if (g_strcmp0(split[1], "debug") == 0) {
+		g_autofree gchar *joined = g_strjoinv(" ", &split[2]);
+		message = g_strdup_printf("Debug: %s", joined);
 	} else {
-		g_print("Unknown command: %s\n", split[1]);
+		message = g_strdup_printf("Unknown handler output: %s", line);
 	}
+
+	g_print("Custom Handler: %s\n", message);
```

Comment:
With this, the extra log lines will be emitted redundantly as now both the `install_args_update` as well as the `g_print` can print the same message. Is this intentional? Also, note that we use g_print() preferably in the frontend for user output while the library functions us  g_message()` and similar.

While the `g_print()` prepends a 'Custom Handler: ' the messages emitted by `install_args_update` do not give a hint about the scope of these messages. Maybe one should add a prefix there, too?
This method is used both for the full custom handler as well as (more frequently used) pre-install and post-install handler.

Reply:
I've removed the print. The message is shown via `install_notify()` in the frontend and via `service_install_notify()` on the service side.